### DISCUSSION TITLE
SimpleStorage ZKSync deployment not working

### DISCUSSION THREAD

**Discussion by freeman237** [Cyfrin/foundry-full-course-cu#2327] [Q&A]
![Screenshot 2024-08-26 190522](https://github.com/user-attachments/assets/ba93f4ab-2eea-463b-980b-e4a82a2c9ec1)
![Screenshot 2024-08-26 190522](https://github.com/user-attachments/assets/e71d47cb-a3ac-4fb2-9742-3f6d8d8ccfa7)

**Comment by EngrPips** [comment]
Hello @freeman237, working with `zksync` can be a bit difficult right now. So if it is not working for you right now you can return to it later. Don't let it slow you down, Patrick stresses it everytime zksync stuffs are optional

**Comment by Haardtripathi** [comment]
Hello @freeman237 , can you please provide me with some more details so that I can help with your issue?
1.Whether you have saved your .sol file inside the contracts directory
2.zksolc Version you are using

Also provide me with the exact error its showing.

**Reply by Payro27** [reply]
Hello @Haardtripathi i have the same issue, im using the 0.8.24 solidity version and i created a file in the default workspace called "Contracts" where i have the SimpleStorage file

**Reply by Payro27** [reply]
with the name of the folder as "contracts" works perfect

**Reply by EngrPips** [reply]
Awesome. Interesting!

**Reply by Haardtripathi** [reply]
@Payro27 Perfect. All the best.

**Comment by JusticeSol** [comment]
@freeman237 All you need to do is uncheck the "Verify contract" box and go ahead to deploy the contract. The verify option is currently having some issues. Goodluck

**Reply by Payro27** [reply]
i have the same problem and i deployed without verifing the contract and i still have the same issue

**Reply by EngrPips** [reply]
Yeah, sometimes it just doesn't work. I used to run into this problem, and I came back later and tried again, and it worked. I can't even explain why that is. Don't let zksync stuff slow you down as it is optional.

**Reply by Payro27** [reply]
Okay ill continue with the course then, ty @EngrPips

**Reply by Payro27** [reply]
@EngrPips now is working, my problem was that i named the folder as "Contracts" instead of "contracts", i changed it and i just deployed the contract in zksync testnet

**Reply by EngrPips** [reply]
Awesome, I am glad you figured this out. Keep crushing it!

**Comment by EngrPips** [comment]
Interesting.

**Reply by JusticeSol** [reply]
Hope it helped you deploy your contract? @EngrPips

**Reply by EngrPips** [reply]
I am not currently doing the section. Actually.

**Reply by JusticeSol** [reply]
Alright, Goodluck.

**Comment by ManuelJG1999** [comment]
I was having the same issue.
Try using a different workspace. Find a folder named "contract", create a new file.sol, copy the code and try compile it.
it should work.

**Reply by EngrPips** [reply]
Thanks for the heads-up.

**Comment by AboloreDev** [comment]
Hello @freeman237, working with zksync can be a bit difficult right now. So if it is not working for you right now you can return to it later. Don't let it slow you down, Patrick stresses it everytime zksync stuffs are optional

also work me through your folder structure.  did you create the simplestorage.sol inside the contract folder before deploying on zksync plugIn

**Reply by EngrPips** [reply]
Indeed
